hostinfo/fileperms.lua

```diff
@@ -0,0 +1,109 @@
+--[[
+Copyright 2015 Rackspace
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+]]--
+local table = require('table')
+local los = require('los')
+local async = require('async')
+local fs = require('fs')
+local fmt = require('string').format
+
+--[[ Check file permissions ]]--
+local HostInfo = require('./base').HostInfo
+local Info = HostInfo:extend()
+function Info:initialize()
+  HostInfo.initialize(self)
+end
+
+function Info:run(callback)
+  if los.type() ~= 'linux' then
+    self._error = 'Unsupported OS for file permissions'
+    callback()
+    return
+  end
+
+  local fileList = {
+  '/etc/grub.conf',
+  '/boot/grub/grub.cfg',
+  '/etc/passwd',
+  '/etc/shadow',
+  '/etc/hosts.allow',
+  '/etc/hosts.deny',
+  '/etc/anacrontab',
+  '/etc/crontab',
+  '/etc/cron.hourly',
+  '/etc/cron.daily',
+  '/etc/cron.weekly',
+  '/etc/cron.monthly',
+  '/etc/cron.d',
+  '/etc/ssh/sshd_config',
+  '/etc/gshadow',
+  '/etc/group',
+  '/etc/login.defs',
+  '/var/run/php-fpm.sock'
+}
+local octalFilePerms = {}
+local stickyBit = {}
+local sgid = {}
+local suid = {}
+
+async.forEachLimit(fileList, 5, function(file, cb)
+  fs.exists(file, function(err, data)
+    if err then
+      --[[This error may occasionally warn us of missing files, we can ignore it or use it upstream]]--
+      self._error = fmt('fs.exists in fileperms.lua erred: %s', err)
+      return cb()
+    end
+    if data then
+      fs.stat(file, function(err, stat)
+        if err then
+          self._error = fmt('fs.stat in fileperms.lua erred: %s', err)
+          cb()
+          return
+        end
+        if stat then
+          --[[Convert a 5 number mode to a 3 number octal]]--
+          octalFilePerms[file] = bit.band(stat.mode, tonumber("777", 8))
```

Comment:
Go ahead and make this mask a decimal too to match the others.
`0777` is `511` in decimal.